/*This patch file is from Qualcomm Technologies, Inc. and is provided under the following license:
Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
SPDX-License-Identifier: BSD-3-Clause-Clear
*/
From 5e634cac6aaf867de3aea661d83200ddf6ff4708 Mon Sep 17 00:00:00 2001
From: Ravi Ranjan Kumar <ravirk@qti.qualcomm.com>
Date: Fri, 26 Sep 2025 11:30:00 +0530
Subject: [PATCH] Fix autoconf macro ordering for proper compiler initialization

This patch ensures that AC_PROG_CC is called before AC_LBL_C_INIT_BEFORE_CC,
which resolves potential issues with compiler detection during configuration.

Signed-off-by: Ravi Ranjan Kumar <ravirk@qti.qualcomm.com>
Upstream-Status: Inappropriate [Downstream]
Status: Applied internally; not suitable for upstream due to custom build flow

aclocal.m4     | 1 -
configure.ac   | 1 +
2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/aclocal.m4 b/aclocal.m4
index c67e5727..5178b66d 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -38,7 +38,6 @@ dnl
 AC_DEFUN(AC_LBL_C_INIT_BEFORE_CC,
 [
     AC_BEFORE([$0], [AC_LBL_C_INIT])
-    AC_BEFORE([$0], [AC_PROG_CC])
     AC_BEFORE([$0], [AC_LBL_FIXINCLUDES])
     AC_BEFORE([$0], [AC_LBL_DEVEL])
     AC_ARG_WITH(gcc, [  --without-gcc           don't use gcc])
diff --git a/configure.ac b/configure.ac
index 18d59b6d..e04f2070 100644
--- a/configure.ac
+++ b/configure.ac
@@ -17,6 +17,7 @@ AC_PREREQ(2.69)
 AC_INIT(tcpdump, m4_esyscmd_s([cat VERSION]))
 AC_CONFIG_SRCDIR(tcpdump.c)
 
+AC_PROG_CC
 AC_CANONICAL_HOST
 
 AC_LBL_C_INIT_BEFORE_CC(V_INCLS)
